### PR TITLE
:arrow_up: Update Nancy template to RC1. Fixes #553

### LINKED ISSUE #553: Nancy Generator creates a broken project for ASP.NET vNext
Hi all,

Just dipping my toes into ASP.NET 5 using MsCode, by following this tutorial:

http://www.codeproject.com/Tips/1067038/Getting-Up-and-Running-with-ASP-NET-using-VS-Code

Being the massive NancyFX nut that I am, I decided to try a Nancy project, but I cant seem to get it to work.

I have all the various req's, npm, node, git and I've set up the dnx/vnext command line environment.

    C:\Users\shawty\aspnet5demo\NancyTest>dnvm list
    
    Active Version           Runtime Architecture OperatingSystem Alias
    ------ -------           ------- ------------ --------------- -----
      *    1.0.0-rc1-update1 clr     x86          win             default
           1.0.0-rc1-update1 coreclr x64          win
    
    C:\Users\shawty\aspnet5demo\NancyTest>

A dnu restore command pulled down all the required dependencies, and a dnu build, produced the following:

    C:\Users\shawty\aspnet5demo\NancyTest>dnu build
    Microsoft .NET Development Utility Clr-x86-1.0.0-rc1-16231
    
    
    Building NancyTest for DNX,Version=v4.5.1
      Using Project dependency NancyTest 1.0.0
        Source: C:\Users\shawty\aspnet5demo\NancyTest\project.json
    
      Using Package dependency Microsoft.AspNet.FileProviders.Abstractions 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.AspNet.FileProviders.Abstractions\1.0.0-beta8
        File: lib\dnx451\Microsoft.AspNet.FileProviders.Abstractions.dll
    
      Using Package dependency Microsoft.AspNet.FileProviders.Physical 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.AspNet.FileProviders.Physical\1.0.0-beta8
        File: lib\dnx451\Microsoft.AspNet.FileProviders.Physical.dll
    
      Using Package dependency Microsoft.AspNet.Hosting 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.AspNet.Hosting\1.0.0-beta8
        File: lib\dnx451\Microsoft.AspNet.Hosting.dll
    
      Using Package dependency Microsoft.AspNet.Hosting.Abstractions 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.AspNet.Hosting.Abstractions\1.0.0-beta8
        File: lib\dnx451\Microsoft.AspNet.Hosting.Abstractions.dll
    
      Using Package dependency Microsoft.AspNet.Hosting.Server.Abstractions 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.AspNet.Hosting.Server.Abstractions\1.0.0-beta8
        File: lib\dnx451\Microsoft.AspNet.Hosting.Server.Abstractions.dll
    
      Using Package dependency Microsoft.AspNet.Http 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.AspNet.Http\1.0.0-beta8
        File: lib\dnx451\Microsoft.AspNet.Http.dll
    
      Using Package dependency Microsoft.AspNet.Http.Abstractions 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.AspNet.Http.Abstractions\1.0.0-beta8
        File: lib\dnx451\Microsoft.AspNet.Http.Abstractions.dll
    
      Using Package dependency Microsoft.AspNet.Http.Extensions 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.AspNet.Http.Extensions\1.0.0-beta8
        File: lib\dnx451\Microsoft.AspNet.Http.Extensions.dll
    
      Using Package dependency Microsoft.AspNet.Http.Features 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.AspNet.Http.Features\1.0.0-beta8
        File: lib\dnx451\Microsoft.AspNet.Http.Features.dll

      Using Package dependency Microsoft.AspNet.Owin 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.AspNet.Owin\1.0.0-beta8
        File: lib\dnx451\Microsoft.AspNet.Owin.dll
    
      Using Package dependency Microsoft.AspNet.Server.Kestrel 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.AspNet.Server.Kestrel\1.0.0-beta8
        File: lib\dnx451\Microsoft.AspNet.Server.Kestrel.dll
    
      Using Package dependency Microsoft.AspNet.WebUtilities 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.AspNet.WebUtilities\1.0.0-beta8
        File: lib\dnx451\Microsoft.AspNet.WebUtilities.dll
    
      Using Package dependency Microsoft.Dnx.Compilation.Abstractions 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Dnx.Compilation.Abstractions\1.0.0-beta8
        File: lib\dnx451\Microsoft.Dnx.Compilation.Abstractions.dll
    
      Using Package dependency Microsoft.Dnx.Runtime.Abstractions 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Dnx.Runtime.Abstractions\1.0.0-beta8
        File: lib\dnx451\Microsoft.Dnx.Runtime.Abstractions.dll
    
      Using Package dependency Microsoft.Framework.Configuration 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Framework.Configuration\1.0.0-beta8
        File: lib\dnx451\Microsoft.Framework.Configuration.dll
    
      Using Package dependency Microsoft.Framework.Configuration.Abstractions 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Framework.Configuration.Abstractions\1.0.0-beta8
        File: lib\dnx451\Microsoft.Framework.Configuration.Abstractions.dll
    
      Using Package dependency Microsoft.Framework.Configuration.CommandLine 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Framework.Configuration.CommandLine\1.0.0-beta8
        File: lib\dnx451\Microsoft.Framework.Configuration.CommandLine.dll
    
      Using Package dependency Microsoft.Framework.Configuration.EnvironmentVariables 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Framework.Configuration.EnvironmentVariables\1.0.0-beta8
        File: lib\dnx451\Microsoft.Framework.Configuration.EnvironmentVariables.dll
    
      Using Package dependency Microsoft.Framework.Configuration.FileExtensions 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Framework.Configuration.FileExtensions\1.0.0-beta8
        File: lib\dnx451\Microsoft.Framework.Configuration.FileExtensions.dll
    
      Using Package dependency Microsoft.Framework.Configuration.Json 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Framework.Configuration.Json\1.0.0-beta8
        File: lib\dnx451\Microsoft.Framework.Configuration.Json.dll
    
      Using Package dependency Microsoft.Framework.DependencyInjection 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Framework.DependencyInjection\1.0.0-beta8
        File: lib\dnx451\Microsoft.Framework.DependencyInjection.dll
    
      Using Package dependency Microsoft.Framework.DependencyInjection.Abstractions 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Framework.DependencyInjection.Abstractions\1.0.0-beta8
        File: lib\dnx451\Microsoft.Framework.DependencyInjection.Abstractions.dll
    
      Using Package dependency Microsoft.Framework.Logging 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Framework.Logging\1.0.0-beta8
        File: lib\dnx451\Microsoft.Framework.Logging.dll
    
      Using Package dependency Microsoft.Framework.Logging.Abstractions 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Framework.Logging.Abstractions\1.0.0-beta8
        File: lib\dnx451\Microsoft.Framework.Logging.Abstractions.dll
    
      Using Package dependency Microsoft.Framework.Primitives 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Framework.Primitives\1.0.0-beta8
        File: lib\dnx451\Microsoft.Framework.Primitives.dll
    
      Using Package dependency Microsoft.Framework.WebEncoders.Core 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Framework.WebEncoders.Core\1.0.0-beta8
        File: lib\dnx451\Microsoft.Framework.WebEncoders.Core.dll
    
      Using Package dependency Microsoft.Net.Http.Headers 1.0.0-beta8
        Source: C:\Users\shawty\.dnx\packages\Microsoft.Net.Http.Headers\1.0.0-beta8
        File: lib\dnx451\Microsoft.Net.Http.Headers.dll
    
      Using Package dependency Nancy 1.3.0
        Source: C:\Users\shawty\.dnx\packages\Nancy\1.3.0
        File: lib\net40\Nancy.dll
    
      Using Package dependency Nancy.Owin 1.4.1
        Source: C:\Users\shawty\.dnx\packages\Nancy.Owin\1.4.1
        File: lib\net40\Nancy.Owin.dll
    
      Using Package dependency Newtonsoft.Json 6.0.6
        Source: C:\Users\shawty\.dnx\packages\Newtonsoft.Json\6.0.6
        File: lib\net45\Newtonsoft.Json.dll
    
      Using Package dependency OWIN 1.0.0
        Source: C:\Users\shawty\.dnx\packages\OWIN\1.0.0
        File: lib\net40\Owin.dll
    
      Using Package dependency System.Diagnostics.Tracing.Telemetry 4.0.0-beta-23409
        Source: C:\Users\shawty\.dnx\packages\System.Diagnostics.Tracing.Telemetry\4.0.0-beta-23409
        File: lib\dotnet\System.Diagnostics.Tracing.Telemetry.dll
    
      Using Package dependency System.Globalization 4.0.0
        Source: C:\Users\shawty\.dnx\packages\System.Globalization\4.0.0
    
      Using Package dependency System.Numerics.Vectors 4.1.1-beta-23409
        Source: C:\Users\shawty\.dnx\packages\System.Numerics.Vectors\4.1.1-beta-23409
        File: ref\dotnet\System.Numerics.Vectors.dll
    
      Using Package dependency System.Resources.ResourceManager 4.0.0
        Source: C:\Users\shawty\.dnx\packages\System.Resources.ResourceManager\4.0.0
    
      Using Package dependency System.Runtime 4.0.0
        Source: C:\Users\shawty\.dnx\packages\System.Runtime\4.0.0
    
      Using Package dependency System.Runtime.Extensions 4.0.0
        Source: C:\Users\shawty\.dnx\packages\System.Runtime.Extensions\4.0.0

      Using Package dependency System.Threading 4.0.0
        Source: C:\Users\shawty\.dnx\packages\System.Threading\4.0.0
    
      Using Assembly dependency fx/mscorlib 4.0.0
        Source: C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.1\mscorlib.dll
    
      Using Assembly dependency fx/System 4.0.0
        Source: C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.1\System.dll
    
      Using Assembly dependency fx/System.Core 4.0.0
        Source: C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.1\System.Core.dll
    
      Using Assembly dependency fx/Microsoft.CSharp 4.0.0
        Source: C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.1\Microsoft.CSharp.dll
    
      Using Assembly dependency fx/System.Runtime 4.0.10
        Source: C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.1\Facades\System.Runtime.dll
    
      Using Assembly dependency fx/System.Collections.Concurrent 4.0.0
        Source: C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.1\Facades\System.Collections.Concurrent.dll
    
    
    Build succeeded.
        0 Warning(s)
        0 Error(s)
    
    Time elapsed 00:00:02.3994429
    Total build time elapsed: 00:00:02.4303250
    Total projects built: 1

As far as I can see, everything works fine, but when I attempt to run the app I consistantly get

    C:\Users\shawty\aspnet5demo\NancyTest>dnx web
    System.InvalidOperationException: No service for type 'Microsoft.Dnx.Runtime.IApplicationEnvironment' has been registered.
       at Microsoft.Framework.DependencyInjection.ServiceProviderExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
       at Microsoft.Framework.DependencyInjection.ServiceProviderExtensions.GetRequiredService[T](IServiceProvider provider)
       at Microsoft.AspNet.Hosting.Program.Main(String[] args)
       at Microsoft.AspNet.Server.Kestrel.Program.Main(String[] args)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at Microsoft.Dnx.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly, String[] args, IServiceProvider serviceProvider)
       at Microsoft.Dnx.ApplicationHost.Program.<>c__DisplayClass3_0.<ExecuteMain>b__0()
       at System.Threading.Tasks.Task`1.InnerInvoke()
       at System.Threading.Tasks.Task.Execute()

I've no idea what's happening here (Partly due to the fact I'm still just experimenting with vNext, and i'm not using vs2015 to do it either, I'm doing it all at the CLI.

all the other project types work fine, and i've tried this on a couple of different machines, with the same outcome.

The machines are all Win7 x64 SP1 based builds.

Cheers
Shawty

### PR DESCRIPTION
**Pull request by peterblazejewicz**
This commit updates Nancy template to support RC1.
- project dependencies update to RC1
- Nancy package update to latest version
- support new hosting model from RC1
- small white-space fixes

This PR passes tests, builds and run succesfully on RC1 installed on OS X.

/cc
@spboyer 
Mind reviewing?

Thanks!

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
@jchannon 
Mind reviewing this? TIA

File: templates/projects/nancy/project.json [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+    },
     "tooling": {
-      "defaultNamespace": "<%= namespace %>"
+        "defaultNamespace": "<%= namespace %>"
     },
     "dependencies": {
-        "Microsoft.AspNet.Server.Kestrel": "1.0.0-beta8",
-        "Microsoft.AspNet.Owin": "1.0.0-beta8",
-        "Nancy": "1.3.0"
+        "Microsoft.AspNet.IISPlatformHandler": "1.0.0-rc1-final",
```
Comment (jchannon) [inline templates/projects/nancy/project.json]: No need for IIS?

Comment (peterblazejewicz) [inline templates/projects/nancy/project.json]: OK, IIS ref removed. Thanks!

**Comment by jchannon** [comment]
:+1

**Comment by peterblazejewicz** [comment]
Thanks!

**Pull request merged**